File: packages/dry/fixtures/semantic-classes/src/billing.py
class BillingTotals:
    def compute(self, lines):
        goods = 0
        taxes = 0
        for line in lines:
            extended = line["quantity"] * line["unit_price"]
            goods += extended
            taxes += extended * line["tax_rate"]
        freight = 0 if goods > 250 else 15
        coupon = 25 if goods > 400 else 0
        return goods + taxes + freight - coupon

File: packages/dry/fixtures/semantic-classes/src/billing.ts
export class BillingTotals {
  compute(lines: ReadonlyArray<{ quantity: number; unitPrice: number; taxRate: number }>): number {
    let goods = 0;
    let taxes = 0;
    for (const line of lines) {
      const extended = line.quantity * line.unitPrice;
      goods += extended;
      taxes += extended * line.taxRate;
    }
    const freight = goods > 250 ? 0 : 15;
    const coupon = goods > 400 ? 25 : 0;
    return goods + taxes + freight - coupon;
  }
}
